### PR TITLE
per document timeout

### PR DESCRIPTION
**Pull request by cristicbz**

### CODE REVIEW

File: exe/ExampleMain.hs [hunk 1]
```diff
@@ -130,11 +134,29 @@ parseHandler tzs = do
...
+        emptyEntitiesOnTimeout :: ((Text, Integer), Text) -> Maybe [Entity] -> Snap [Entity]
+        emptyEntitiesOnTimeout (_, text) Nothing = do
+          logError
+            $ Text.encodeUtf8
+            $ Text.concat [
+              "Timed out when parsing entities in document: ```"
+              , text
+              , "```."
+            ]
+          return []
```
Comment (tommilligan) [inline exe/ExampleMain.hs]: FYI, `logError` appears to evaluate character escape sequences here - in the example I ran, it evaluated newlines like `\n\n` in the JSON request and split the log over several lines. Not sure if this is what you intended.

**Review by tommilligan — APPROVED** [review]
Happy the timeout works well - tested with `["short text", "long text", "short text"]`, the long document timed out and returned `[]` entities, everything else worked as normal.

The builtin `Snap` timeout appears to be 60 seconds - are we happy that we won't get 6 comment timeouts in the same batch?

File: exe/ExampleMain.hs (hunk 1 above)
Comment (cristicbz) [reply exe/ExampleMain.hs]: Not what I intended no, but I also don't know how to fix it---I'll have a look, but if not, I'll just have to live with it.

Comment (cristicbz) [reply exe/ExampleMain.hs]: Right, so the problem isn't that it evaluates newlines in the JSON request, but that the JSON request itself has newline chars. I changed the code to do `show text` (the equivalent of `repr`) in Python.

**Pull request merged**